h1st/tuner/hyperparameter_tuner.py

```diff
@@ -0,0 +1,278 @@
+            else:
+                raise ValueError("value of", name_, "was not provided.")
+        return config_space
+
+    def _get_config_space_grid(self, parameters):
+        config_space = {}
+        for param in parameters:
+            name_, _, min_, max_, choice_ = self._get_values_from_dict(param)
+            if len(choice_) != 0:
+                config_space[name_] = tune.grid_search(choice_)
+            elif (min_ is not None) or (max_ is not None):
+                raise ValueError(f"{name_}: grid search does not support range input")
+            else:
+                raise ValueError(f"value of {name_} was not provided")
+        return config_space
+
+    def _get_config_space_random(self, parameters):
+        config_space = {}
+        for param in parameters:
+            name_, type_, min_, max_, choice_ = self._get_values_from_dict(param)
+            if (min_ is not None) and (max_ is not None):
+                if type_ == "float":
+                    config_space[name_] = tune.uniform(min_, max_)
+                elif type_ == "int":
+                    config_space[name_] = tune.randint(min_, max_)
+                else:
+                    raise TypeError(name_, ": range input cannot be string type")
+            elif len(choice_) != 0:
+                config_space[name_] = tune.choice(choice_)
+            else:
+                raise ValueError(f"value of {name_} was not provided")
+        return config_space
+
+    def _get_values_from_dict(self, param):
+        name_ = param.get('name')
+        type_ = param.get('type')
+        min_ = param.get('min')
+        max_ = param.get('max')
+        choice_ = param.get('choice', [])
+        return (name_, type_, min_, max_, choice_)
+
+    def _create_h1st_model_trainable(self, model_class, parameters, target_metric):
+        class H1stModelTrainable(tune.Trainable):
+            def setup(self, config):
+                h1st.init()
+                self.config = config
+                self.hyperparameter = parameters
+                self.timestep = 0
+                self.kwargs = {}
+                for param in self.hyperparameter:
+                    k = param["name"]
+                    self.kwargs[k] = int(round(self.config[k])) if param["type"]=="int" \
+                                        else float(self.config[k]) if param["type"]=="float" \
+                                        else self.config[k]
+                self.h1_ml_model = model_class(**self.kwargs)
+                DATA_LOCK_DIR = "./data_lock"
+                lock_file = f"{DATA_LOCK_DIR}/data.lock"
+                if not os.path.exists(DATA_LOCK_DIR):
+                    os.makedirs(DATA_LOCK_DIR)
+                with FileLock(os.path.expanduser(lock_file)):
+                    data = self.h1_ml_model.load_data()
+                self.prepared_data = self.h1_ml_model.prep(data)
+
+            def step(self):
+                self.timestep += 1
```

Comment:
self.h1_ml_model is repeated so many times, looks busy. Please refactor @TgithubJ